validate/validate.go

```diff
@@ -490,6 +499,30 @@ func namespaceValid(ns rspec.Namespace) bool {
 	return true
 }
 
+func nestedValid(os, pathA, pathB string) bool {
+	var purePathA, purePathB string
```

Comment:
The paths were passed by value, so you drop `purePath*` and just clobber `pathA` and `pathB`.